monai/networks/layers/simplelayers.py

```diff
@@ -411,7 +411,7 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         Args:
             x: in shape [Batch, chns, H, W, D].
         """
-        _kernel = [gaussian_1d(s, truncated=self.truncated, approx=self.approx) for s in self.sigma]
+        _kernel = [gaussian_1d(s, truncated=self.truncated, approx=self.approx).to(x.device) for s in self.sigma]
```

Comment:
I think it's the user's responsibility to make sure this module `GaussianFilter` and `x` are on the same device, following pytorch nn.modules' behaviour

Reply:
Unfortunately `ensure_tuple_rep(sigma, int(spatial_dims))` converts to numpy. Therefore, any input that was on the gpu will be copied to the cpu.

Reply:
I see, I feel we need a flag in `ensure_tuple_rep` to control the conversion...

Reply:
Yes you're right, I should have just enhanced the underlying functionality. I'll do that instead.